packages/lambda-powertools-middleware-correlation-ids/__tests__/alb.js

```diff
@@ -0,0 +1,102 @@
+const _ = require('lodash')
+const uuid = require('uuid/v4')
+const { invokeHandler } = require('./lib')
+
+global.console.log = jest.fn()
+
+const alb = require('./event-templates/alb.json')
+const genAlbEvent = (correlationIds = {}) => {
+  const event = _.cloneDeep(alb)
+  event.headers = correlationIds
+  return event
+}
+
+describe('Correlation IDs middleware (ALB)', () => {
+  describe('when sampleDebugLogRate = 0', () => {
+    it('always sets debug-log-enabled to false', () => {
+      const requestId = uuid()
+      const event = genAlbEvent()
+      invokeHandler(event, requestId, 0, x => {
+        expect(x['awsRequestId']).toBe(requestId)
+        expect(x['x-correlation-id']).toBe(requestId)
+        expect(x['debug-log-enabled']).toBe('false')
+      })
+    })
+  })
+
+  describe('when sampleDebugLogRate = 1', () => {
+    it('always sets debug-log-enabled to true', () => {
+      const requestId = uuid()
+      const event = genAlbEvent()
+      invokeHandler(event, requestId, 1, x => {
+        expect(x['awsRequestId']).toBe(requestId)
+        expect(x['x-correlation-id']).toBe(requestId)
+        expect(x['debug-log-enabled']).toBe('true')
+      })
+    })
+  })
+
+  describe('when correlation ID is not provided in the event', () => {
+    it('sets it to the ALB Request ID', () => {
```

Comment:
does this test still apply? since ALB doesn't give you a request ID

Reply:
or maybe it should be renamed to reflect that it gets set to the Lambda request ID

Reply:
probably should be renamed. I'll update.